Disable editing in UI for KW default policies

aalves08 opened the pull request:
Fixes #682 

- Remove `"Edit Config"` and `"Edit YAML"` in UI for KW default policies
- Add `"Edit policy settings"` in UI as table action for KW default policies, which will take the user to the `kubewarden-defaults` chart installation screen
- `Update mode` on KW default policies should take the user to the `kubewarden-defaults` chart installation screen
- Add a table column to CAP's list view called `Source`, so that we can distinguish KW default policies from the rest
- Change the `clone`  table action on default KW policies to **NOT** copy annotations that make them a default policy (went with `his.metadata?.labels?.['app.kubernetes.io/name']` which is a default KW policy equals `kubewarden-defaults`)

<img width="1738" alt="Screenshot 2024-04-18 at 16 22 58" src="https://github.com/rancher/kubewarden-ui/assets/97888974/5cd36928-dc27-4eef-8d5e-db8f948e1887">
<img width="1738" alt="Screenshot 2024-04-18 at 16 23 03" src="https://github.com/rancher/kubewarden-ui/assets/97888974/9449850d-6903-4d1d-897b-3f2e73e69937">

https://github.com/rancher/kubewarden-ui/assets/97888974/d311af80-2729-42e2-86bc-b844e83349e0

kravciak commented:
Hi, I started playing with this PR, my notes so far (ordered by priority):
- [ ] edit action redirects me to wrong version of defaults helm chart (rc3 instead of 1.9.5-rc3)
- [ ] remove/disable Edit Policy Settings for policies with no settings
  they are not visible on helm chart edit
- [ ] remove/disable delete button for recommended (default) policies?
  deleted policy are recreated on kubewarden upgrade
- [ ] missing warning that UI 1.4.2 requires with kubewarden 1.12
  otherwise edit action points to helm chart with no questions
- [ ] can edit action send directly to policy settings tab?
  ..&version=1.9.5-rc3`#no-privilege-escalation%20policy%20settings` for example
- [ ] clone of recommended policy does not show settings
- [ ] when I clone policy action button is called `Action` instead of `Create` or `Finish`

aalves08 commented:
@kravciak latest updates: 

- fixed bug with edit policy settings version capturing rc text 
- update source naming for CAPs 
- added banner to CAPs list to inform user that they should update kubewarden-defaults chart (unfortunately I cannot disable the `Edit Policy Settings` from the table actions based on the version check, since it's totally bound to the JS class of a resource and I cannot run the code to do the assertion there... If this is considered important, please open an issue on kubewarden-ui, but the fix will need to go into Rancher Dashboard)

<img width="1758" alt="Screenshot 2024-04-19 at 18 13 33" src="https://github.com/rancher/kubewarden-ui/assets/97888974/9dee3dc6-d793-4624-87b1-e3132e127cd3">
